pkg/playback/http_playback.go

```diff
@@ -0,0 +1,660 @@
+}
+
+func newRecordServer(remoteURL string, webhookURL string) (httpRecorder recordServer) {
+	httpRecorder = recordServer{}
+	httpRecorder.remoteURL = remoteURL
+	httpRecorder.webhookURL = webhookURL
+
+	return httpRecorder
+}
+
+func (httpRecorder *recordServer) insertCassette(writer io.Writer) error {
+	recorder, err := newInteractionRecorder(writer)
+	if err != nil {
+		return err
+	}
+	httpRecorder.recorder = recorder
+
+	return nil
+}
+
+func (httpRecorder *recordServer) webhookHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("\n[WEBHOOK] --> STRIPE %v to %v --> POST to %v", r.Method, r.RequestURI, httpRecorder.webhookURL)
+
+	// --- Pass request to local webhook endpoint
+	resp, err := forwardRequest(r, httpRecorder.webhookURL)
+
+	// TODO: this response is going back to Stripe, what is the correct error handling logic here?
+	if err != nil {
+		handleErrorInHandler(w, fmt.Errorf("Unexpected error forwarding webhook to client: %w", err), 500)
+		return
+	}
+
+	// --- Write response back to client
+	// The header *must* be written first, since writing the body with implicitly and irreversibly set
+	// the status code to 200 if not already set.
+	// Copy header
+	w.WriteHeader(resp.StatusCode)
+	copyHTTPHeader(w.Header(), resp.Header)
+
+	// Copy body
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		handleErrorInHandler(w, fmt.Errorf("Unexpected error processing client webhook response: %w", err), 500)
+		return
+	}
+	io.Copy(w, bytes.NewBuffer(bodyBytes))
+	resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+	defer resp.Body.Close()
+
+	err = httpRecorder.recorder.write(incomingInteraction, newSerializableHTTPRequest(r), newSerializableHTTPResponse(resp))
+	if err != nil {
+		handleErrorInHandler(w, fmt.Errorf("Unexpected error writing webhook interaction to cassette: %w", err), 500)
+		return
+	}
+}
+
+func (httpRecorder *recordServer) handler(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("\n--> %v to %v", r.Method, r.RequestURI)
+
+	// --- Pass request to remote
+	var resp *http.Response
+	var err error
+
+	resp, err = forwardRequest(r, httpRecorder.remoteURL+r.RequestURI)
+
+	if err != nil {
+		handleErrorInHandler(w, err, 500)
+		return
+	}
+	fmt.Printf("\n<-- %v from %v\n", resp.Status, strings.ToUpper(httpRecorder.remoteURL))
+
+	// --- Write response back to client
+	// The header *must* be written first, since writing the body with implicitly and irreversibly set
+	// the status code to 200 if not already set.
+	// Copy header
+	w.WriteHeader(resp.StatusCode)
+	copyHTTPHeader(w.Header(), resp.Header)
+
+	// Copy body
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close() // we need to close the body
+
+	if err != nil {
+		handleErrorInHandler(w, err, 500)
+		return
+	}
+	io.Copy(w, bytes.NewBuffer(bodyBytes))
+	// we need to reset the reader to be able to read the body again
+	resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
```

Comment:
This feels strange -- we shouldn't have to read from resp.Body more than once.
Would it be possible to make "newSerializableHTTPResponse" accept an io.Reader instead of an http.Response? I don't think it's strictly necessary that what we "serialize" is a literal http.Response... it just has to be something that we can construct an http.Response from.

(If this doesn't make sense, I probably don't know what I'm talking about)

Reply:
I guess what I am suggesting is, don't serialize/deserialize the http request/response themselves -- serialize a 
```
struct {
  headers: map[string]string
  body: []byte
  statusCode: int
}
```
and then add a little bit more wrapping code around this to convert to the literal http.Request / http.Response when you need to...

Reply:
(this still might not make sense)

Reply:
This makes sense! And I implemented this fix in: 9d9fdc6f020ceb8f36df74c454aee83b82dc9c82